tests/dogtag/shared/java/ca/generateCRMFRequest.java

```diff
@@ -59,11 +59,9 @@ public boolean generate() {
             return false;
         }
         PrintStream ps = null;
-        try {
-            ps = new PrintStream(new FileOutputStream(output_file));
+        try (ps = new PrintStream(new FileOutputStream(output_file))) {
             ps.println(cert_request);
             ps.flush();
```

Comment:
The `ps` variable declaration can be moved inside the `try-with-resource`. The `ps.flush()` is probably redundant since IIUC closing the stream will flush the stream too.

Reply:
> The `ps` variable declaration can be moved inside the `try-with-resource`. 

Yeah, I had kinda half done it for one - good spot this would not have compiled, I have fixed it.

> The `ps.flush()` is probably redundant since IIUC closing the stream will flush the stream too.

I had the same thought but IIRC `AutoCloseable` doesn't put any restriction on how it's implementations implement `close()`, so I left it explicit. If the stream type were to be changed in the future we may accidentally lose the `flush()` (assuming that these streams do actually do it).